### PR TITLE
Update boondoggle

### LINKED ISSUE #344: Get boondoggle up to date so that the example script works
The purpose of this task is to get boondoggle's source up to date with quilc and the qvm s.t. the example script in the README runs successfully.

### PR DESCRIPTION
**Pull request by cbosborn**
Worked with @ecpeterson to get boondoggle to run successfully. Resolves #344.

### CODE REVIEW

File: boondoggle/src/producers.lisp [hunk 1]
```diff
@@ -136,9 +136,9 @@ PARAMETER-BOUNDS is a list of maximum random values for the gate parameters."
                (unless random-gate-flag
                  (mapcar #'random (gate-set-record-parameter-bounds gate-record))))
              (gate-invocation (make-instance 'quil::gate-application
-                                             :operator gate-name
-                                             :arguments (mapcar #'quil::qubit qubit-indices)
-                                             :parameters (mapcar #'quil::constant gate-parameters))))
+                                             :operator (quil::named-operator gate-name)
```
Comment (notmgsk) [inline boondoggle/src/producers.lisp]: `named-operator` is exported so you can get away with a single colon. Small nit pick.

Comment (notmgsk) [reply boondoggle/src/producers.lisp]: Likewise with `qubit` and `constant` below.

File: boondoggle/src/processors.lisp [hunk 2]
```diff
@@ -46,6 +46,12 @@
 
 (defmethod apply-process ((processor processor-measures) &rest data)
   (let ((data (quil::copy-instance (first data))))
+    (setf (quil::parsed-program-memory-definitions data)
+      (append (list (quil::make-memory-descriptor
+		     :name "ro"
+		     :type quil::quil-bit
+		     :length (length (elt (quil::chip-specification-objects (processor-measures-chip-specification processor)) 0))))
```
Comment (appleby) [inline boondoggle/src/processors.lisp]: Not certain, but iiuc I think this `(length (elt ...))` bit can be shorted to `(quil::chip-spec-n-qubits (processor-measures-chip-specification processor))`

File: boondoggle/src/consumers.lisp [hunk 3]
```diff
@@ -45,7 +45,8 @@
               (with-output-to-string (s)
                 (quil::print-parsed-program parsed-program s)))
             (classical-addresses
-              (get-consumer-registers consumer parsed-program))
+              (a:plist-hash-table
+	       (list "ro" (get-consumer-registers consumer parsed-program))))
```
Comment (appleby) [inline boondoggle/src/consumers.lisp]: Indentation here looks off-by-one. The open paren should be indented one extra space.

File: boondoggle/src/processors.lisp [hunk 4]
```diff
@@ -46,6 +46,12 @@
 
 (defmethod apply-process ((processor processor-measures) &rest data)
   (let ((data (quil::copy-instance (first data))))
+    (setf (quil::parsed-program-memory-definitions data)
+      (append (list (quil::make-memory-descriptor
+		     :name "ro"
```
Comment (appleby) [inline boondoggle/src/processors.lisp]: ditto here re: indentation

File: boondoggle/src/consumers.lisp (hunk 3 above)
Comment (appleby) [reply boondoggle/src/consumers.lisp]: Huh. Maybe just github being weird with the diff. In this view it looks like it's off-by-two... 🤷‍♂

File: boondoggle/src/processors.lisp (hunk 4 above)
Comment (notmgsk) [reply boondoggle/src/processors.lisp]: I think @cbosborn has committed the cardinal sin and is using tabs. 😱

Comment (appleby) [reply boondoggle/src/processors.lisp]: Still weird that github doesn't display them consistently... Maybe we should add a rigetti-standard `.dir-locals.el` to the project root.  Maybe even get crazy with an `.editorconfig`, for all the poor lost souls using Visual Studio Code (TM)

Comment (cbosborn) [reply boondoggle/src/processors.lisp]: @notmgsk I noticed it too and was wondering, but you helped me set up emacs so I'm punting this one back to you :)

File: boondoggle/src/consumers.lisp (hunk 3 above)
Comment (cbosborn) [reply boondoggle/src/consumers.lisp]: Yes also bothering me. Let me get it resolved

File: boondoggle/src/producers.lisp [hunk 5]
```diff
@@ -136,9 +136,9 @@ PARAMETER-BOUNDS is a list of maximum random values for the gate parameters."
                (unless random-gate-flag
                  (mapcar #'random (gate-set-record-parameter-bounds gate-record))))
              (gate-invocation (make-instance 'quil::gate-application
```
Comment (ecpeterson) [inline boondoggle/src/producers.lisp]: I know we were making minimal changes as we went through this, but this `make-instance` block now has a preferred synonym in quilc: `(apply #'build-gate gate-name (coerce gate-parameters 'list) (coerce qubit-indices 'list))`.

**Pull request merged**